src/tabset/tabset.ts

```diff
@@ -0,0 +1,80 @@
+import {
+  Component,
+  Input,
+  ContentChildren,
+  QueryList,
+  Directive,
+  TemplateRef,
+  ContentChild,
+  ViewRef,
+  ViewContainerRef,
+  AfterContentChecked
+} from 'angular2/core';
+
+// TODO: remove when https://github.com/angular/angular/commit/f4e6994634a9cf6c2c20aa8c2268233e2667edf8 is released
+@Directive({selector: '[ngbTemplateOutlet]'})
+export class NgbTemplateOutlet {
+  private _insertedViewRef: ViewRef;
+
+  constructor(private _viewContainerRef: ViewContainerRef) {}
+
+  @Input()
+  set ngbTemplateOutlet(templateRef: TemplateRef) {
+    if (this._insertedViewRef) {
+      this._viewContainerRef.remove(this._viewContainerRef.indexOf(this._insertedViewRef));
+    }
+
+    if (templateRef) {
+      this._insertedViewRef = this._viewContainerRef.createEmbeddedView(templateRef);
+    }
+  }
+}
+
+@Directive({selector: 'template[ngb-tab-title]'})
+export class NgbTabTitle {
+  constructor(public templateRef: TemplateRef) {}
+}
+
+@Directive({selector: 'template[ngb-tab-content]'})
+export class NgbTabContent {
+  constructor(public templateRef: TemplateRef) {}
+}
+
+@Directive({selector: 'ngb-tab'})
+export class NgbTab {
+  @Input() title: string;
+
+  @ContentChild(NgbTabContent) contentTpl: NgbTabContent;
+  @ContentChild(NgbTabTitle) titleTpl: NgbTabTitle;
+}
+
+@Component({
+  selector: 'ngb-tabset',
+  template: `
+    <ul class="nav nav-tabs" role="tablist">
+      <li class="nav-item" *ngFor="#tab of tabs; #i=index">
+        <a class="nav-link" [class.active]="i === activeIdx" (click)="select(i)">
+          {{tab.title}}<template [ngbTemplateOutlet]="tab.titleTpl?.templateRef"></template>
```

Comment:
Yup, this is "by design", see: https://github.com/ng-bootstrap/core/pull/181/files#diff-f4f98e2db742c4772ab831ae9adb2dacR69

Having said this I don't mind either way so if you / anyone else prefers other logic, please let me know.